charms/worker/k8s/src/charm.py

```diff
@@ -601,6 +601,44 @@ def _get_scrape_jobs(self):
             log.exception("Failed to get COS token.")
         return []
 
+    @on_error(
+        ops.WaitingStatus("Sharing Cluster Version"),
+        AssertionError,
+    )
```

Comment:
are there other exception types you want to allow to escape as unit errors?  If not -- the simple approach is this one
```suggestion
    @on_error(ops.WaitingStatus("Sharing Cluster Version"))
```